sdk/keyvault/keyvault/inc/az_keyvault.h

```diff
@@ -5,43 +5,103 @@
 #define AZ_KEYVAULT_H
 
 #include <az_contract.h>
+#include <az_credential.h>
+#include <az_http_pipeline.h>
+#include <az_http_request_builder.h>
 #include <az_result.h>
 #include <az_span.h>
+#include <az_str.h>
```

Comment:
please clean up includes in the header files. Anything that is used only in the corresponding .c file - you can include in the .c file, no need to put it in the header. For instance, I don't think you need az_str.h and http_request_builder